### PR TITLE
Fix webassets path in Makefile

### PR DESCRIPTION
**Pull request by kimlisa**
#### Problems
- When i ran `make release`, the enterprise was being built with OSS web assets, b/c the directory used to store the `webassets.zip` wasn't being deleted before building enterprise (which skips building enterprise web assets if this file exists, and uses what existed which is the OSS web assets zip)
- `make full` command adds OSS web assets with whatever existed in the previous zip before (could be enterprise web assets stuff or older oss web assets stuff)
- `make full-ent` had same problem with `make release`'s enterprise 

#### Solution
- For `make release`, there was a naming mistake, so i fixed that
- Everything else, I deleted the web assets build zip file before zipping

### CODE REVIEW

**Review by webvictim — APPROVED** [review]
LGTM!

**Review by awly — COMMENTED** [review]
Can we keep enterprise webassets in a separate directory, like before?

For example, make a copy of https://github.com/gravitational/teleport/blob/master/lib/web/static_embed.go that's built with an `embed_webassets_e` tag and embeds the different `webassets.zip`?

deleting and re-zipping it is fine, but a bit wasteful

**Comment by kimlisa** [comment]
@awly but we have to delete the older zip file anyways, otherwise zip will add whatever was in the previous zip file along with newer contents (if any). At least this is the behavior I saw while testing it out. 

So if we had a existing zip file, and webassets got updated (updates contain files with diff hash numbers), the zip will include the old files with the new files.

**Comment by awly** [comment]
Ah, I guess we don't have the logic to detect "did webassets change and should the archive be re-created?".
I suppose it's fine to re-zip assets on every build.

File: Makefile [hunk 1]
```diff
@@ -144,7 +144,9 @@ endif
 .PHONY:full-ent
 full-ent:
 ifneq ("$(OS)", "windows")
-	@if [ -f e/Makefile ]; then $(MAKE) -C e full; fi
+	@if [ -f e/Makefile ]; then \
+	rm -fr $(ASSETS_BUILDDIR)/webassets.zip; \
```
Comment (awly) [inline Makefile:148]:
```suggestion
	rm $(ASSETS_BUILDDIR)/webassets.zip; \
```

File: Makefile [hunk 2]
```diff
@@ -401,15 +403,15 @@ tag:
 
 # build/webassets.zip archive contains the web assets (UI) which gets
 # appended to teleport binary
-$(ASSETS_BUILDDIR)/webassets.zip: | $(ASSETS_BUILDDIR)
 $(ASSETS_BUILDDIR)/webassets.zip: ensure-webassets
```
Comment (awly) [inline Makefile:406]:
```suggestion
$(ASSETS_BUILDDIR)/webassets.zip: ensure-webassets $(ASSETS_BUILDDIR)
```

File: Makefile [hunk 3]
```diff
@@ -401,15 +403,15 @@ tag:
...
 ifneq ("$(OS)", "windows")
 	@echo "---> Building OSS web assets."
+	rm -fr $(ASSETS_BUILDDIR)/webassets.zip
+	mkdir -p $(ASSETS_BUILDDIR)
 	cd webassets/teleport/ ; zip -qr ../../$@ .
 endif
 
-$(ASSETS_BUILDDIR):
-	mkdir -p $@
```
Comment (awly) [inline Makefile:412]: revert this, no need to re-create the directory on every build

File: Makefile [hunk 4]
```diff
@@ -401,15 +403,15 @@ tag:
 
 # build/webassets.zip archive contains the web assets (UI) which gets
 # appended to teleport binary
-$(ASSETS_BUILDDIR)/webassets.zip: | $(ASSETS_BUILDDIR)
 $(ASSETS_BUILDDIR)/webassets.zip: ensure-webassets
 ifneq ("$(OS)", "windows")
 	@echo "---> Building OSS web assets."
+	rm -fr $(ASSETS_BUILDDIR)/webassets.zip
+	mkdir -p $(ASSETS_BUILDDIR)
```
Comment (awly) [inline Makefile:410]:
```suggestion
	rm $(ASSETS_BUILDDIR)/webassets.zip
```

File: Makefile (hunk 1 above)
Comment (kimlisa) [reply Makefile]: removed flags

File: Makefile (hunk 2 above)
Comment (kimlisa) [reply Makefile]: revised

**Review by awly — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Pull request merged**